pkg/schema/loader.go

```diff
@@ -0,0 +1,192 @@
+func (l *GoSchemaLoader) LoadSchema(ctx context.Context, opts GoLoaderOpts) GoSchema {
+	return NewGoSchema(
+		opts.SchemaName, opts.GroupName, opts.GroupVersion, opts.SchemaOpenapi, opts.SchemaObject, opts.SchemaListObject,
+	)
+}
+
+// ThemaSchemaLoader loads Thema schemas.
+type ThemaSchemaLoader struct {
+	lib thema.Library
+}
+
+// ThemaLoaderOpts are the options for loading Thema schemas.
+type ThemaLoaderOpts struct {
+	SchemaFS         fs.FS
+	SchemaPath       string
+	SchemaVersion    thema.SyntacticVersion
+	GroupName        string                          // TODO: should be part of the lineage
+	GroupVersion     string                          // TODO: should be part of the lineage
+	SchemaOpenapi    apiextensionsv1.JSONSchemaProps // TODO: should be part of the lineage
+	SchemaType       interface{}
+	SchemaObject     runtime.Object
+	SchemaListObject runtime.Object
+}
+
+// ProvideThemaSchemaLoader returns a new Thema schema loader, which uses Thema library lib for registering schemas.
+func ProvideThemaSchemaLoader(lib thema.Library) *ThemaSchemaLoader {
+	return &ThemaSchemaLoader{
+		lib: lib,
+	}
+}
+
+// LoadThemaSchema loads a new Thema schema with options opts.
+func (l *ThemaSchemaLoader) LoadSchema(ctx context.Context, opts ThemaLoaderOpts) (*ThemaSchema, error) {
+	lin, err := loadThemaLineage(opts.SchemaPath, opts.SchemaFS, l.lib)
+	if err != nil {
+		return nil, err
+	}
+
+	// Calling this ensures our program cannot start,
+	// if the Go DataSource.Model type is not aligned with the canonical schema version in the lineage.
+	if _, err := newJSONKernel(lin, opts.SchemaPath, opts.SchemaType); err != nil {
+		return nil, err
+	}
+
+	zsch, err := lin.Schema(opts.SchemaVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := thema.AssignableTo(zsch, opts.SchemaType); err != nil {
+		return nil, err
+	}
+
+	return NewThemaSchema(
+		lin, opts.GroupName, opts.GroupVersion, opts.SchemaOpenapi, opts.SchemaObject, opts.SchemaListObject,
+	), nil
+}
+
+func loadThemaLineage(path string, cueFS fs.FS, lib thema.Library, opts ...thema.BindOption) (thema.Lineage, error) {
+	prefix := filepath.FromSlash(path)
+	fs, err := prefixWithGrafanaCUE(prefix, cueFS)
+	if err != nil {
+		return nil, err
+	}
+	inst, err := load.InstancesWithThema(fs, prefix)
+
+	// Need to trick loading by creating the embedded file and
+	// making it look like a module in the root dir.
+	if err != nil {
+		return nil, err
+	}
+
+	val := lib.Context().BuildInstance(inst)
+
+	lin, err := thema.BindLineage(val, lib, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	return lin, nil
+}
+
+func newJSONKernel(lin thema.Lineage, loaderPath string, object interface{}) (kernel.InputKernel, error) {
+	return kernel.NewInputKernel(kernel.InputKernelConfig{
+		Lineage:     lin,
+		Loader:      kernel.NewJSONDecoder(loaderPath),
+		To:          thema.SV(0, 0),
+		TypeFactory: func() interface{} { return object },
+	})
+}
```

Comment:
Seems like we're not actually using this now, so might as well just drop it. (Can easily put it back if/when it makes sense to do so.)
```suggestion
```